src/TrackGenerator.cpp

```diff
@@ -413,6 +413,19 @@ void TrackGenerator::setNumThreads(int num_threads) {
   omp_set_num_threads(_num_threads);
   if (_geometry != NULL)
     _geometry->reserveKeyStrings(num_threads);
+
+  /* Print CPU assignments, useful for NUMA where by-socket is the preferred
+   * CPU grouping */
+  std::vector<int> cpus;
+#pragma omp parallel for schedule(static,1)
+  for (int i=0; i<num_threads; i++)
+    cpus.push_back(sched_getcpu());
+  std::stringstream str_cpus;
+  for (int i=0; i<num_threads; i++)
+      str_cpus << cpus.at(i) << " ";
+
+  if (num_threads > 1)
+    log_printf(NORMAL, "CPUs on rank 0 process: %s", str_cpus.str().c_str());
```

Comment:
Is it OK to use NODAL output?

Reply:
Yes, good idea. I avoided it at first because I am going to run hundreds of MPI processes but I think I ll just deal with the logs.